src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -419,6 +405,11 @@ public Path getJavaAppEngineSdkPath() {
     return getSdkPath().resolve(JAVA_APPENGINE_SDK_PATH);
   }
 
+  @VisibleForTesting
+  Path getJavaBinaryPath() {
+    return javaHomePath.resolve("bin/java");
```

Comment:
should this just return javaHomePath.resolve("bin/java").getRealPath()?  It seems like you're always canonicalize it from the caller.